### PR TITLE
Add documentation for using SendGrid and for Mailgun API

### PR DESCRIPTION
**Pull request by paskal**
As suggested in #655, I've added documentation for email providers:
- Mailgun usage via API
- SendGrid usage via API and SMTP

I've tested all configurations before writing about them.

### CODE REVIEW

**Comment by ttys3** [comment]
@paskal  LGTM

my side container almost done:  https://github.com/ttys3/smtp-brd

**Comment by paskal** [comment]
@ttys3 thank you! Please ping me as soon as you'll have docker-compose example, I will test it and include into documentation.

File: docs/email.md [hunk 1]
```diff
@@ -31,39 +31,128 @@ SMTP_TIMEOUT
...
-      - SMTP_HOST=smtp.eu.mailgun.org
-      - SMTP_PORT=465
-      - SMTP_TLS=true
-      - SMTP_USERNAME=postmaster@mg.example.com
-      - SMTP_PASSWORD=secretpassword
-      - AUTH_EMAIL_FROM=notify@example.com
+
+#### API
+
+Using SMTP directly as in example above exposes sending server IP address. Alternative
```
Comment (umputun) [inline docs/email.md:46]: I think the part about "exposes sending server IP address" should be clarified a little bit. As far as I recall this "exposing" is a part of one of the SMTP RFCs and not a side effect of the particular mailgun relay usage.

Comment (paskal) [reply docs/email.md]: Clarified a little, please check the current version.

**Pull request merged**